homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,320 @@
+        """Return the minimum temperature."""
+        match self._attr_hvac_mode:
+            case HVACMode.COOL:
+                attribute = clusters.Thermostat.Attributes.AbsMinCoolSetpointLimit
+            case HVACMode.HEAT | HVACMode.HEAT_COOL:
+                attribute = clusters.Thermostat.Attributes.AbsMinHeatSetpointLimit
+            case _:
+                return DEFAULT_MIN_TEMP
+
+        if (value := self._get_temperature_in_degrees(attribute)) is not None:
+            return value
+        return DEFAULT_MIN_TEMP
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        match self._attr_hvac_mode:
+            case HVACMode.HEAT:
+                attribute = clusters.Thermostat.Attributes.AbsMaxHeatSetpointLimit
+            case HVACMode.COOL | HVACMode.HEAT_COOL:
+                attribute = clusters.Thermostat.Attributes.AbsMaxCoolSetpointLimit
+            case _:
+                return DEFAULT_MAX_TEMP
+
+        if (value := self._get_temperature_in_degrees(attribute)) is not None:
+            return value
+        return DEFAULT_MAX_TEMP
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        if self._attr_hvac_mode == HVACMode.COOL:
+            return self._get_temperature_in_degrees(
+                clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
+            )
+        return self._get_temperature_in_degrees(
+            clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
+        )
+
+    @property
+    def target_temperature_high(self) -> float | None:
+        """Return the highbound target temperature we try to reach."""
+        if self._attr_hvac_mode == HVACMode.HEAT_COOL:
+            return self._get_temperature_in_degrees(
+                clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
+            )
+        return None
+
+    @property
+    def target_temperature_low(self) -> float | None:
+        """Return the lowbound target temperature we try to reach."""
+        if self._attr_hvac_mode == HVACMode.HEAT_COOL:
+            return self._get_temperature_in_degrees(
+                clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
+            )
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        target_hvac_mode: HVACMode | None = kwargs.get(ATTR_HVAC_MODE)
+        if target_hvac_mode is not None:
+            await self.async_set_hvac_mode(target_hvac_mode)
+
+        current_mode = target_hvac_mode or self.hvac_mode
+        command = None
+        if current_mode in (HVACMode.HEAT, HVACMode.COOL):
+            # when current mode is either heat or cool, the temperature arg must be provided.
+            temperature = kwargs.get(ATTR_TEMPERATURE)
+            if temperature is None:
+                raise ValueError("Temperature must be provided")
```

Comment:
When I write this test case, I always get intercepted by voluptuous `voluptuous.error.MultipleInvalid: expected float for dictionary value @ data['temperature']`